src/analysis/inner_product_space/cross_product.lean

```diff
@@ -0,0 +1,167 @@
+the orientation and inner product structure. The partial evaluation of the volume form Ω on `u` and
+`v` gives an element Ω(u, v, ⬝) of the dual of `E`.  This can then be identified with `E` itself
+using the canonical isometry (induced by the inner product) between `E` and its dual.
+
+See `matrix.cross_product` for a construction of the cross-product in co-ordinates. -/
+def cross_product : E →ₗ[ℝ] E →ₗ[ℝ] E :=
+begin
+  let to_dual : E ≃ₗ[ℝ] (E →ₗ[ℝ] ℝ) :=
+    (inner_product_space.to_dual ℝ E).to_linear_equiv ≪≫ₗ linear_map.to_continuous_linear_map.symm,
+  let z : alternating_map ℝ E ℝ (fin 0) ≃ₗ[ℝ] ℝ :=
+    alternating_map.const_linear_equiv_of_is_empty.symm,
+  let y : alternating_map ℝ E ℝ (fin 1) →ₗ[ℝ] E →ₗ[ℝ] ℝ :=
+    (linear_map.llcomp ℝ E (alternating_map ℝ E ℝ (fin 0)) ℝ z) ∘ₗ
+      alternating_map.curry_left_linear_map,
+  let y' : alternating_map ℝ E ℝ (fin 1) →ₗ[ℝ] E :=
+    (linear_map.llcomp ℝ (alternating_map ℝ E ℝ (fin 1)) (E →ₗ[ℝ] ℝ) E to_dual.symm) y,
+  let u : alternating_map ℝ E ℝ (fin 2) →ₗ[ℝ] E →ₗ[ℝ] E :=
+    (linear_map.llcomp ℝ E (alternating_map ℝ E ℝ (fin 1)) _ y') ∘ₗ
+      alternating_map.curry_left_linear_map,
+  exact u ∘ₗ (alternating_map.curry_left_linear_map ω.volume_form),
+end
+
+local infixl (name := cross_product) ` ×₃ `: 74 := ω.cross_product
+
+/-- Defining property of the cross-product: for vectors `u`, `v`, `w`, the inner product of u × v
+with w is equal to ω(u, v, w), where -/
+lemma inner_cross_product_apply (u v w : E) : ⟪u ×₃ v, w⟫ = ω.volume_form ![u, v, w] :=
+by simp only [cross_product, linear_equiv.trans_symm, linear_equiv.symm_symm,
+  linear_isometry_equiv.to_linear_equiv_symm, alternating_map.curry_left_linear_map_apply,
+  linear_map.coe_comp, function.comp_app, linear_map.llcomp_apply,
+  linear_equiv.coe_coe, linear_equiv.trans_apply, linear_isometry_equiv.coe_to_linear_equiv,
+  linear_isometry_equiv.norm_map, submodule.coe_norm,
+  inner_product_space.to_dual_symm_apply, alternating_map.curry_left_apply_apply,
+  alternating_map.const_linear_equiv_of_is_empty_symm_apply,
+  eq_self_iff_true, linear_map.coe_to_continuous_linear_map', matrix.zero_empty]
+
+attribute [irreducible] cross_product
+
+/-- The cross-product is antisymmetric: the cross-product of a vector with itself is 0. -/
+lemma cross_product_apply_self (v : E) : v ×₃ v = 0 :=
+begin
+  apply ext_inner_right ℝ,
+  intros w,
+  rw [inner_cross_product_apply, inner_zero_left],
+  refine ω.volume_form.map_eq_zero_of_eq ![v, v, w] _ (_ : (0 : fin 3) ≠ 1),
+  { simp },
+  { norm_num }
+end
+
+/-- The cross-product of `u` and `v` is orthogonal to `u`. -/
+lemma inner_cross_product_apply_self (u v : E) : ⟪u ×₃ v, u⟫ = 0 :=
+begin
+  rw ω.inner_cross_product_apply u v u,
+  refine ω.volume_form.map_eq_zero_of_eq ![u, v, u] _ (_ : (0 : fin 3) ≠ 2),
+  { simp },
+  { norm_num }
+end
+
+/-- The cross-product of `u` and `v` is orthogonal to `v`. -/
+lemma inner_cross_product_apply_apply_self (u v : E) : ⟪u ×₃ v, v⟫ = 0 :=
```

Comment:
I don't understand the name of this lemma, but more importantly this follows from `u ×₃ v = - v ×₃ u`, which is a lemma that should be there I feel.

Reply:
I added the swap lemma.  It's a little bit of a pain to deduce the "repeated argument equals zero" versions from the "swap negates" versions, because `eq_neg_self_iff` is for rings, not torsion-free groups. (I could swear I added the latter version to mathlib at some point, couldn't track it down ...). So I didn't refactor the other proof to use it.